Enabled user to specify the reverse geocode service

kengorton opened the pull request:
Users might publish their own Reverse Geocode Service. and wish to use it instead of the ArcGIS Online Reverse Geocode service.

Also upgraded to 10.6.

kengorton committed:
Enabled user to specify the reverse geocode service

Users might publish their own Reverse Geocode Service. and wish to use it instead of the ArcGIS Online Reverse Geocode service.

hanoch commented on reverse-geocoder-processor/src/main/java/com/esri/geoevent/processor/reversegeocoder/ReverseGeocoderProcessor.java:
```diff
@@ -139,7 +146,9 @@ private GeoEvent processGeoEvent(GeoEvent geoEvent) throws MalformedURLException
 		// The default response format is html.
 		agolSearchFormat = "json"; 
-		URL agolURL = new URL("http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/reverseGeocode?location=" + Double.toString(lon) + "," + Double.toString(lat) + "&distance=" + Integer.toString(agolSearchDistance) + "&outSR=" + Integer.toString(wkid) + "&f=" + agolSearchFormat);
+		//URL agolURL = new URL("http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/reverseGeocode?location=" + Double.toString(lon) + "," + Double.toString(lat) + "&distance=" + Integer.toString(agolSearchDistance) + "&outSR=" + Integer.toString(wkid) + "&f=" + agolSearchFormat);
```
do we need to keep this as a comment here?

hanoch commented on pom.xml:
```diff
@@ -4,7 +4,7 @@
 	<groupId>com.esri.geoevent.parent</groupId>
 	<artifactId>reverse-geocoder</artifactId>
-	<version>10.5.0</version>
+	<version>10.6.0</version>
```
let's keep upgrading the version to 10.6.0 separate from this PR so we can include this enhancement with 10.5.x

hanoch commented on reverse-geocoder-processor/pom.xml:
```diff
@@ -5,7 +5,7 @@
 		<groupId>com.esri.geoevent.parent</groupId>
 		<artifactId>reverse-geocoder</artifactId>
-		<version>10.5.0</version>
+		<version>10.6.0</version>
```
let's keep upgrading the version to 10.6.0 separate from this PR so we can include this enhancement with 10.5.x

hanoch commented on reverse-geocoder-processor/src/main/java/com/esri/geoevent/processor/reversegeocoder/ReverseGeocoderProcessorDefinition.java:
```diff
@@ -74,7 +78,7 @@ public String getDomain()
 	public String getVersion()
 	{
-		return "10.5.0";
+		return "10.6.0";
```
let's keep upgrading the version to 10.6.0 separate from this PR so we can include this enhancement with 10.5.x

hanoch commented on reverse-geocoder-processor/src/main/java/com/esri/geoevent/processor/reversegeocoder/ReverseGeocoderProcessorDefinition.java:
```diff
@@ -44,6 +45,9 @@ public ReverseGeocoderProcessorDefinition()
 			PropertyDefinition agolSearchDistanceProperty = new PropertyDefinition(AGOL_SEARCHDISTANCE_PROPERTY, PropertyType.Integer, 100, "ArcGIS Online Reverse Geocoder Search Radius (m)", "The distance from the geoevent's location within which a matching address should be searched.", true, false);
 			propertyDefinitions.put(AGOL_SEARCHDISTANCE_PROPERTY, agolSearchDistanceProperty);
 
+			PropertyDefinition geocodeServiceUrlProperty = new PropertyDefinition(GEOCODE_SERVICE_PROPERTY, PropertyType.String, "http://geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/reverseGeocode", "ArcGIS Online Reverse Geocoder Search Radius (m)", "The distance from the geoevent's location within which a matching address should be searched.", true, false);
```
we will need to eventually i18n these strings

kengorton committed:
rolled back to versino 10.5

kengorton committed:
Updated to reflect capabilities of the processor

Updated to reflect new capabilities of the processor to allow users to specify the reverse geocode service.

hanoch reviewed (approved):
Need to i18n this repo at some point.

hanoch committed:
Merge branch 'master' into use-custom-geocode-service

# Conflicts:
#	README.md
